### ISSUE TITLE
WebSocket Response, write to server/client

### ISSUE THREAD

**Issue by Amiflon** [ProxymanApp/Proxyman#2016]
## Description

Add an input field to the Response window, WebSocket tab, from which you can send data to the server or client while the connection is active. I often work with WebSockets and I miss this feature.
![WebSocket Response](https://github.com/ProxymanApp/Proxyman/assets/4671644/fe107754-b51c-460b-be26-dfae04941333)

## Why this feature/change is important?

This will take Proxyman to the next level without having to use other software, such as Insomnia.

**Comment by NghiaTranUIT** [comment]
May I ask:
- Do you know what app allows you to write a new WS/WSS during the interception?

Insomnia/Postman is different because they make a WS/WSS Connection, so it's important to be able to write the WS Message.

**Comment by Amiflon** [comment]
> May I ask:
> 
> * Do you know what app allows you to write a new WS/WSS during the interception?
> 
> Insomnia/Postman is different because they make a WS/WSS Connection, so it's important to be able to write the WS Message.

No, I don't know. Does Proxyman work with WS/WSS by using the principle of (Server) < - > ((Client)) Proxyman ((Server)) < - > (Client)? If so, why not implement it?

**Comment by Amiflon** [comment]
Manipulating WebSocket messages with Burp Suite: [https://portswigger.net/burp/documentation/desktop/testing-workflow/websockets/manipulating-websocket-messages]()

They write that "whistle" also has this feature, but I need to check it:
[https://github.com/avwo/whistle?tab=readme-ov-file]()

**Comment by NghiaTranUIT** [comment]
Ah, it's allowed you to repeat the WS/WSS and send a new message.

It's different what you initially mentioned that you'd like to add new WS Message during the interception.

**Comment by jason-fbg** [comment]
Has there been any movement on this? It would be a valuable tool

**Comment by NghiaTranUIT** [comment]
@jason-fbg 

> Right-click a message that you want to re-send or modify (for example, an outbound chat message) and select Send to Repeater. Burp creates a new WebSocket tab in Repeater.

It seems the Burp Suite doesn't allow you to modify the Websocket Message on the fly. They only allow you to repeat the website by creating a new websocket connection. 

To clarify, it create a new websocket, and you can manually type the message or copy it from the source

Is it what you're looking for @jason-fbg 🤔

**Comment by jason-fbg** [comment]
Hey @NghiaTranUIT, from what I'm reading (although I am skimming) it does seem to be more than that but I have also found [mitmproxy](https://docs.mitmproxy.org/stable/addons/examples/#websocket-inject-message) may be able to do it too.

**Comment by NghiaTranUIT** [comment]
May I ask what your use case is? You need to:
1. Modify the websocket on-the-fly without interrupting the connection
2. Just repeat a message by creating a new websocket connection, and be able to manually send the websocket

**Comment by jason-fbg** [comment]
It would be 1. for my usecase
